src/Cassandra.Tests/ConnectionTests.cs

```diff
@@ -310,17 +310,56 @@ public async Task ReadParse_While_Disposing_Faults_Tasks_But_Never_Throws()
             await TestHelper.WaitUntilAsync(() => totalFrames == responses.Count, 100, 30).ConfigureAwait(false);
             Assert.AreEqual(totalFrames, responses.Count);
         }
+        
+        [Test]
+        public void Should_HandleDifferentProtocolVersionsInDifferentConnections_When_OneConnectionResponseVersionIsDifferentThanSerializer()
```

Comment:
Good coverage!

Reply:
Yeah I copy pasted this test into `master` and it fails so I guess it accomplishes its goal but the reason of the failure is a bit weird. Basically it parses the body length as a ridiculously large number due to the header size being different than expected and the `Task` of the request is never set as `Complete` because of that. 

This makes me wonder if we should make sure that we respect the version that comes in the header even if we are past the `Startup` phase... Theoretically it should never happen but there's no reason not to use the version in the header to figure out the remaining length of the header we are parsing. This would allow us to properly recognize what's going on, log a warning and maybe set the result of the request as a `ProtocolError`? 

I'm going to merge this PR but we can continue discussing this here or offline. After the discussion I'll capture this into a ticket if we think it's worth it to do so.